dataplane/globalbase.cpp

```diff
@@ -1635,41 +1635,60 @@ inline uint64_t generation::count_real_connections(uint32_t counter_id)
 	return (sessions_created - sessions_destroyed + sessions_created_gc - sessions_destroyed_gc) / dataPlane->numaNodesInUse;
 }
 
-void generation::evaluate_service_ring()
+balancer_real_id_t* generation::evaluate_service_ring_one(balancer_real_id_t* start,
+                                                          const balancer_real_id_t* const do_not_exceed,
+                                                          const balancer_service_t& service,
+                                                          balancer_service_range_t& range)
 {
-	balancer_service_ring_t* ring = &balancer_service_ring;
-	uint32_t weight_pos = 0;
-	for (uint32_t service_idx = 0; service_idx < balancer_services_count; ++service_idx)
+	for (uint32_t real_idx = service.real_start;
+	     real_idx < service.real_start + service.real_size;
+	     ++real_idx)
 	{
-		balancer_service_t* service = balancer_services + balancer_active_services[service_idx];
-
-		balancer_service_range_t* range = ring->ranges + balancer_active_services[service_idx];
+		uint32_t real_id = balancer_service_reals[real_idx];
+		balancer_real_state_t* state = balancer_real_states + real_id;
```

Comment:
`balancer_real_state_t* state = &balancer_real_states[real_id];` is **much** better in my opinion.
It immediately conveys the meaning that it's the address of an object in an array, rather than adding a uint32_t value to a pointer (which we don't even know unless we check the definition of balancer_real_states)